PiT 25.1: flow-server no longer provides jakarta.servlet-api as transitive dependency, breaking VaadinSession API

### Description of the bug

In Vaadin 25.1, `flow-server` no longer pulls `jakarta.servlet-api` as a transitive dependency. However, Flow's public API (`VaadinSession.setAttribute()`) internally references `jakarta.servlet.http.HttpSessionBindingListener`. This means any project that uses `VaadinSession` without Spring Boot (which provides the servlet API) will fail to compile.

This affects projects using Jetty directly (like `addon-template`) where the servlet API is provided by the Jetty plugin at runtime but not available at compile time.

```
[ERROR] ThemeChangeSupport.java:[42,24] cannot access jakarta.servlet.http.HttpSessionBindingListener
  class file for jakarta.servlet.http.HttpSessionBindingListener not found
```

Since `VaadinSession` is part of Flow's public API, `flow-server` should either:
1. Keep `jakarta.servlet-api` as a transitive dependency (as it was in 25.0.x), or
2. Ensure its public API does not leak servlet types to callers

### Expected behavior

Code using `VaadinSession.setAttribute()` should compile without the user having to manually declare `jakarta.servlet-api`.

### Minimal reproducible example

```bash
git clone https://github.com/vaadin/addon-template.git
cd addon-template
# Bump to 25.1.0-beta1
perl -pi -e 's|(<vaadin.version>)[^<]+|${1}25.1.0-beta1|' pom.xml
# Try to run (fails at test-compile)
mvn -ntp -B
```

The error occurs in `ThemeChangeSupport.java` which calls `VaadinSession.setAttribute()`.

### Versions

- Vaadin / Flow version: 25.1.0-beta1 (works fine with 25.0.6)

--- a/pom.xml
+++ b/pom.xml
@@ -50,6 +50,13 @@
             <groupId>com.vaadin</groupId>
             <!-- Consider limiting the dependencies to flow-server only something, but for most end users depending on vaadin-core is ok -->
             <artifactId>vaadin-core</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.servlet</groupId>
+            <artifactId>jakarta.servlet-api</artifactId>
+            <version>6.1.0</version>
+            <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -50,7 +50,6 @@
             <groupId>com.vaadin</groupId>
             <!-- Consider limiting the dependencies to flow-server only something, but for most end users depending on vaadin-core is ok -->
             <artifactId>vaadin-core</artifactId>
-            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>jakarta.servlet</groupId>
@@ -279,5 +278,3 @@
         </profile>
     </profiles>
 </project>
-
-

